incubator/hnc/pkg/controllers/object_controller_new.go

```diff
@@ -38,25 +59,306 @@ type ObjectReconcilerNew struct {
 
 	// GVK is the group/version/kind handled by this reconciler.
 	GVK schema.GroupVersionKind
+
+	// Affected is a channel of event.GenericEvent (see "Watching Channels" in
+	// https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches.html) that is used to
+	// enqueue additional objects that need updating.
+	Affected chan event.GenericEvent
 }
 
 // +kubebuilder:rbac:groups=*,resources=*,verbs=get;list;watch;create;update;patch;delete
 
 func (r *ObjectReconcilerNew) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	resp := ctrl.Result{}
+	ctx := context.Background()
 	log := r.Log.WithValues("trigger", req.NamespacedName)
-	log.Info("New object controller placeholder")
-	return resp, nil
+
+	log.Info("Reconciling object")
+
+	// Read the object. If it's missing, assume it's been deleted or not yet created.
+	inst := &unstructured.Unstructured{}
+	inst.SetGroupVersionKind(r.GVK)
+	inst.SetNamespace(req.Namespace)
+	inst.SetName(req.Name)
+	found := true
+	if err := r.Get(ctx, req.NamespacedName, inst); err != nil {
+		if !errors.IsNotFound(err) {
+			log.Info("Couldn't read")
+			return resp, err
+		}
+		found = false
```

Comment:
Instead of an explicit variable, you _could_ also use something like `inst.CreationTimestamp.IsZero()` inside the function (see https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/controllers/hierarchy_controller.go#L452).

Reply:
I used `found := inst.GetCreationTimestamp() != v1.Time{}` since the `inst` is a pointer to `unstructured.Unstructured`.

Reply:
That's weird. It works fine for GetCreationTimestamp here: https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/controllers/hierarchy_controller.go#L452. The dot operator should work fine both on structs and pointers in Go. Do you have any idea why it's not working for you? Are those runtime or compile time errors?

Reply:
They are compile time errors.

I think the example `inst *api.HierarchyConfiguration` in hierarchy controller has `v1.ObjectMeta` with `CreationTimestamp` field. While for `inst *unstructured.Unstructured`, `GetCreationTimestamp()` returns a non-addressable value:

```
func (u *Unstructured) GetCreationTimestamp() metav1.Time {
	var timestamp metav1.Time
	timestamp.UnmarshalQueryParameter(getNestedString(u.Object, "metadata", "creationTimestamp"))
	return timestamp
}
```
Maybe that's the difference? I found something at:
https://stackoverflow.com/questions/44543374/cannot-take-the-address-of-and-cannot-call-pointer-method-on

Reply:
Filed #292 292